hazelcast/include/hazelcast/client/exception/IException.h

```diff
@@ -83,6 +83,8 @@ namespace hazelcast {
                  */
                 virtual std::auto_ptr<IException> clone() const;
 
+                virtual bool isProtocolException() const;
```

Comment:
How about making all exceptions protocol Exception and removing this method completely ?
We can give negative error codes to the ones only used locally

Reply:
Yes, changed as suggested. Removed ProtocolException class.